engine/table/src/main/java/io/deephaven/engine/table/impl/TimeTable.java

```diff
@@ -45,49 +46,91 @@
 public class TimeTable extends QueryTable implements Runnable {
     private static final Logger log = LoggerFactory.getLogger(TimeTable.class);
 
-    private static final String TIMESTAMP = "Timestamp";
-    private long lastIndex = -1;
-    private final SyntheticDateTimeSource columnSource;
-    private final TimeProvider timeProvider;
-    private final PerformanceEntry entry;
-    private final boolean isStreaming;
+    public static class Builder {
+        private UpdateSourceRegistrar registrar = UpdateGraphProcessor.DEFAULT;
+        private TimeProvider timeProvider;
+        private DateTime startTime;
+        private long period;
+        private boolean isStreamTable;
 
-    public TimeTable(TimeProvider timeProvider, long period) {
-        this(timeProvider, null, period, false);
-    }
+        public Builder withRegistrar(UpdateSourceRegistrar registrar) {
```

Comment:
`with` and `as` imply to me that you are doing a copy-on-write of an immutable builder.